docs/operate-and-deploy/installation/server-config/index.md

```diff
@@ -310,3 +310,60 @@ Start the ksqlDB server with the configuration file specified.
 <path-to-confluent>/bin/ksql-server-start <path-to-confluent>/etc/ksqldb/ksql-server.properties
 ```
 
+Configuring Listeners for a ksqlDB Cluster
+--------------------------------
+
+Multiple hosts are required to scale ksqlDB processing power and to do that, they must
+form a cluster.  ksqlDB utilizes {{ site.kstreams }} to do this and requires all hosts of
+a cluster to use the same `ksql.service.id`.
+
+```properties
+bootstrap.servers=localhost:9092
+ksql.service.id=my_application_
+```
+
+Once formed, many operations can be run using the client APIs exposed on `listeners`.
+
+In order to utilize pull queries and their high availability functionality,
+the hosts within the cluster must be configured to speak to each other via their
+internal endpoints. The following describes how to configure listeners depending on
+the nature of your environment.
+
+### Internal Network
+
+For the typical setup within an internal network, you might have just one
```

Comment:
I'm a bit confused by the 'internal network' terminology, though it could just be me.

'Internal network' suggests to me that there is an internal and public network, which is more an IaaS setup.  Yet the example is talking more about an example where the `listeners` config is either using some form of `localhost` for the `listeners` config, or where the local host name is not resolvable/routable by other servers running the ksql nodes.

Is this just me? Or would a better name help here?  It seems we have the following setups:

1. Single interface, using resolvable and routable host names, i.e. just using `listeners` config.
2. Single interface, using unresolvable or unroutable host names, i.e. using `listeners` and `advertised` configs.
3. Dual interface, where internal is resolvable and routable host name, i.e. using `listeners` and `internal.listeners`.
4. Dual interface, where internal is unresolvable or unroutable host name, i.e. using `listeners`, `internal.listeners` and `advertised.listeners`.

I think it would make sense to have a section for each and explain the key part of needing to set advertised config is if either:
  - the config used for internal listeners, (either set explicitly or defaulting to `listeners[0]`), is using wildcards (which defaults to localhost) or explicitly using localhost, i.e. so it's not meaningful to other nodes
   - or the explicit host name used for internal listeners, (either set explicitly or defaulting to `listeners[0]`), is either not resolvable to an IP, or that IP is not routable, from other ksql nodes.

The choice between these 4 modes doesn't have that much to do with the users choice between running auth or not running auth, (as you've alluded to in the Unsecure Environment section below). After all, they can run each with auth on or off.   However, _which_ of these configurations they're running may affect how they need to configure auth. Maybe not right now, but certainly once/if we allow different auth to be set on the internal endpoints.  Hence I'd probably change the text in this page to say that running over dual network interfaces makes it possible to configure the network such that the inter-node endpoints ksqlDB uses are not accessible on the public listener(s).  If done correctly, this can be a more secure mode of deployment. And then go one to briefly mention and link to the auth docs as a 'Another security feature you can use is Auth'.

Reply:
I agree that the previous terminology wasn't great.

I swapped out what I had for 4 configurations, as you've outlined above.  I also mentioned the fallback sequence, so it should be intuitive what's in use when configurations are unset.  I kept things fairly simple, so I didn't draw the full decision tree, but I think for documentation, we want to keep it somewhat straightforward.

I also mentioned the dual network interfaces and how this is a similar variation to port filtering.  I agree that this is going to need to be elaborated on a fair amount when we add inter-node auth and this blurb will probably have to be updated as well to at least reference it.